### ISSUE TITLE
[Overlay] Scroll blocking inconsistency on different OS/Browsers

### ISSUE THREAD

**Issue by oktav777** [angular/components#8986]
#### Bug, feature request, or proposal:
bug

#### What is the expected behavior?
Scrolling should be blocked on all platforms when a backdrop overlay is active.

#### What is the current behavior?
There are browsers that don't block scrolling when a backdrop overlay is active.

#### What are the steps to reproduce?
All test are done on the page: https://material.angular.io/components/dialog/overview by opening the modal on clicking `Pick one`.
Also I observed the behavior of the sidebar is different across browsers, thats probably a different issue.

### Windows 10:
* Chrome 62:
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav **has** a scrollbar.
* Opera 49: 
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav **has** a scrollbar.
* Internet Exporer 11:
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav **has** a scrollbar.
* Edge 16: 
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav **hasn't** a scrollbar, sidenav is scrolled with the entire page [the layout of page is broken].
* Firefox 57:
  * Page scrolling **isn't blocked** while modal is opened.
  * Sidenav **hasn't** a scrollbar, sidenav is scrolled with the entire page [the layout of page is broken].

### Android 8.1(Developer preview 2) [nexus 6p]
* Chrome 62:
  * Page scrolling **isn't blocked** while modal is opened.
  * Sidenav scrolls the entire page when reaching the bottom of the sidenav.
  * Page is scrollable on backdrop while sidenav is opened.
* Opera 44:
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav scrolls the entire page when reaching the bottom of the sidenav.
  * Page is scrollable on backdrop while sidenav is opened.
* Firefox 57:
  * Page scrolling **isn't blocked** while modal is opened.
  * Sidenav scrolls the entire page when reaching the bottom of the sidenav.
  * Page is scrollable on backdrop while sidenav is opened.

### iOS 10.3.3 [iPhone 5c]
* Safari 10:
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav scrolls the entire page when reaching the bottom of the sidenav (wait for animation to stop).
  * Page is scrollable on backdrop while sidenav is opened.
* Chrome 62
  * Page scrolling **is blocked** while modal is opened.
  * Sidenav scrolls the entire page when reaching the bottom of the sidenav (wait for animation to stop).
  * Page is scrollable on backdrop while sidenav is opened.

#### What is the use-case or motivation for changing an existing behavior?
The scroll blocking should be consistent among the platforms for a better UX.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
The ones that are used on `material.angular.io`

#### Is there anything else we should know?
All components that use backdrop overlay are affected (ex: menu, select)

**Comment by jturmel** [comment]
Chrome 63 on iOS exhibits this same issue.

**Comment by croso-df** [comment]
Same issue in Chrome 64 Windows. Scrolling not getting blocked.

**Comment by sebastiandenis** [comment]
It looks like the problem is also connected with the viewport size. I have tried with Chrome 64 and page scrolling isn't blocked when viewport size is <=720px. Above 720px scrolling is blocked for Chrome 64.

**Comment by oktav777** [comment]
> It looks like the problem is also connected with the viewport size. I have tried with Chrome 64 and page scrolling isn't blocked when viewport width is <=720px. Above 720px scrolling is blocked for Chrome 64.

I can confirm this!

**Comment by sarora2073** [comment]
+1
Am using dialogs extensively and my app doesn't work on ipad / safari due to this issue. Any touch scrolling actions on dialogs affect the page more often than they perform the desired scrolling on the dialog. This can be confirmed on the 'Dialog with header, scrollable content and actions' example here: https://material.angular.io/components/dialog/overview

**Comment by jelbourn** [comment]
(revisiting older issues as part of a triaging fix-it)

I tried this on:
* Windows (Chrome, Firefox, desktop and mobile view)
* macOS (Safari)
* ChromeOS
* Android (Chrome)

In all cases the scrolling was blocked as expected, so I suspect this was fixed either in the browser or in the library between 2017 and now. Feel free to file a new issue is some problem persists.
